lib/chewy/index/adapter/mongoid.rb

```diff
@@ -0,0 +1,87 @@
+require 'chewy/index/adapter/orm'
+
+module Chewy
+  class Index
+    module Adapter
+      class Mongoid < Orm
+        def self.accepts?(target)
+          defined?(::Mongoid::Document) && (
+            (target.is_a?(Class) && target.ancestors.include?(::Mongoid::Document)) ||
+              target.is_a?(::Mongoid::Criteria))
+        end
+
+        def identify(collection)
+          super(collection).map { |id| id.is_a?(BSON::ObjectId) ? id.to_s : id }
+        end
+
+      private
+
+        def cleanup_default_scope!
+          Chewy.logger.warn('Default type scope order, limit and offset are ignored and will be nullified') if Chewy.logger && sort_or_limit_or_skip_options?
+
+          @default_scope.options.delete(:limit)
+          @default_scope.options.delete(:skip)
+          @default_scope = @default_scope.reorder(nil)
+        end
+
+        def sort_or_limit_or_skip_options?
+          @default_scope.options.values_at(:sort, :limit, :skip).compact.present?
+        end
+
+        def import_scope(scope, options)
+          pluck_in_batches(scope, **options.slice(:batch_size)).map do |ids|
+            yield grouped_objects(default_scope_where_ids_in(ids))
+          end.all?
+        end
+
+        def import_objects(collection, options)
+          direct_import = (default_scope.selector.empty? || @options[:searchable_proc]) &&
+            !options[:raw_import] &&
+            collection.is_a?(Array) &&
+            !collection.empty? &&
+            collection.all? { |item| item.is_a?(::Mongoid::Document) && item.__selected_fields.nil? }
+          options[:direct_import] = direct_import unless options[:direct_import].present?
```

Comment:
This is to maintain parity for direct_import condition.
We don't see this directly in leadgenie but we derive it here based on the args passed to import. 
And then super is called with the right flag

Reply:
For testing, you can run `show_mongo` while calling index.import! contact_objects and see if it makes an unnecessary find query to mongo. Since the contact object is already loaded, it should not make an unnecessary call.